src/config/dnsmasq_config.c

```diff
@@ -442,10 +445,20 @@ bool __attribute__((nonnull(1,3))) write_dnsmasq_config(struct config *conf, boo
 		fputs("\n", pihole_conf);
 	}
 
-	const char *interface = conf->dns.interface.v.s;
+	bool interface_allocated = false;
+	char *interface = conf->dns.interface.v.s;
 	// Use eth0 as fallback interface if the interface is missing
 	if(strlen(interface) == 0)
-		interface = "eth0";
+	{
+		interface = get_gateway_name();
+		if(interface == NULL)
+		{
+			log_err("Unable to get default interface name");
+			interface = (char*)"eth0";
```

Comment:
We already fallback to `eth0` within `get_gateway_name`. Do we need a double-net?